app/components/visibility-setting-list.js

```diff
@@ -0,0 +1,109 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { reads, empty, not } from '@ember/object/computed';
+import KeyboardShortcuts from 'ember-keyboard-shortcuts/mixins/component';
+
+export default Component.extend(KeyboardShortcuts, {
+  classNames: ['visibility-setting-list'],
+
+  keyboardShortcuts: {
+    'esc': 'closeConfirmationModal'
+  },
+
+  // An example item that might be in the options object:
+  // private: {
+  //   displayValue: 'you',
+  //   description: 'Do not allow everyone to see insights from your private builds',
+  //   modalText: 'Do not allow everyone to see my private insights',
+  // }
+  //
+  // The key is used to match selected and currentSelection up with the correct details
+  // `displayValue` is used to generate text for the modal
+  // `description` is for the label next to the radio button
+  // `modalText` can be used to override the generated modal text
+  options: computed(() => ({})),
+  optionKeys: computed('options', function () { return Object.keys(this.options); }),
+  optionValues: computed('options', function () { return Object.values(this.options); }),
```

Comment:
Hmm good point. Yeah it looks like `optionValues` isn't used at all. Although `optionKeys` is used to calculate the index of each item for use in calculating `change`. But maybe there's a better way to do it.

Maybe it would be better if  `options` was an array of objects rather than a single object? Then `options.[]` could be tracked as well as things like `options.@each.{description}` if needed. WDYT?

Reply:
Yes, I agree, array would work better. You could add another property to the option object, like `key` that would contain the former object key, e.g.
```js
{
  key: 'private',
  displayValue: 'you',
  description: 'Do not...',
  modalText: 'Do not...'
}
```
And then aggregate the values by `key`, e.g. `options.findBy('key', 'private')`, `options.mapBy('key')`, etc.

Reply:
Cool, that sounds good to me! Updated this to use an array instead.